### ISSUE TITLE
Remove default argument of Environment template

### ISSUE THREAD

**Issue by Flamefire** [ComputationalRadiationPhysics/picongpu#1594]
There are various instances where `Environment<>::get()` is used to get one of the other singletons (e.g. EventTask). This is dangerous as one might run a 2D simulation and use this method to accidentally get an uninitialized Singleton. E.g. `Environment<>::get().GridController()` would compile w/o any warning/error but crash at runtime, because the 1D-(default)-GridController was never initialized.
There are some singletons that are not templated over the dimensions and safe to use that way, but not all and the difference is not easily visible.

I would remove the default template argument, so `Environment<>::get()` is invalid code and extract all safe-to-get singleton getters into a new class `BaseEnvironment` that is not templated and contains only those non-templated getters as static methods. So `Environment<>::get().Manager()` gets replaced by `BaseEnvironment::.Manager()` which can be done by Search&Replace automatically.

**Comment by psychocoderHPC** [comment]
Yes this was also our initial idea to implement this how you describe it. But our developer leaved use :-(

**Comment by ax3l** [comment]
@psychocoderHPC is this solved with #1890?
